test/functional/test_framework/ellsq.py

```diff
@@ -103,35 +104,47 @@ def r(x,y,i):
 
 def encode(P):
     while True:
-        u = field_random()
-        T = curve_negate(f(u))
-        Q = curve_add(T, P)
-        if is_infinity(Q): Q = T
+        u = fe(random.randrange(1, SECP256K1_ORDER))
+        fe1 = f(u)
+        # convert fe1 to jacobian form for EC operations
+        fe1 = (fe1[0].val, fe1[1].val, 1)
+        T = SECP256K1.negate(fe1)
+        Q = SECP256K1.add(T, P)
+        if SECP256K1.on_curve(Q) is None:
```

Comment:
Add a new `SECP256k1.is_infinity` function instead of overloading `on_curve`, and use that.

Reply:
Done.